lib/features/search/email/presentation/search_email_controller.dart

```diff
@@ -282,36 +251,92 @@ class SearchEmailController extends BaseController
         }
       }
     );
+
+    emailUIActionWorker = ever(
+      mailboxDashBoardController.emailUIAction,
+      (action) {
+        if (action is RefreshChangeEmailAction) {
+          _refreshEmailChanges(newState: action.newState);
+        }
+      },
+    );
   }
 
-  void _onSearchTextInputListener() {
-    if (textInputSearchFocus.hasFocus) {
-      searchIsRunning.value = false;
+  void _refreshEmailChanges({jmap.State? newState}) {
+    log('SearchEmailController::_refreshEmailChanges(): newState: $newState');
+    if (accountId == null ||
+        session == null ||
+        mailboxDashBoardController.currentEmailState == null ||
+        newState == null ||
+        searchIsRunning.isFalse) {
+      return;
     }
+
+    _webSocketQueueHandler?.enqueue(WebSocketMessage(
+      newState: newState,
+      accountId: accountId!,
+      session: session!,
+    ));
+  }
+
+  void _initWebSocketQueueHandler() {
+    _webSocketQueueHandler = WebSocketQueueHandler(
+      processMessageCallback: _handleWebSocketMessage,
+      onErrorCallback: onError,
+    );
   }
 
-  void _refreshEmailChanges() {
-    if (searchIsRunning.isTrue && session != null && accountId != null) {
+  Future<void> _handleWebSocketMessage(WebSocketMessage message) async {
+    try {
+      if (mailboxDashBoardController.currentEmailState == null ||
+          mailboxDashBoardController.currentEmailState == message.newState) {
+        log('SearchEmailController::_handleWebSocketMessage:Skipping redundant state: ${message.newState}');
+        return Future.value();
+      }
+
       final limit = listResultSearch.isNotEmpty
-          ? UnsignedInt(listResultSearch.length)
-          : ThreadConstants.defaultLimit;
+        ? UnsignedInt(listResultSearch.length)
+        : ThreadConstants.defaultLimit;
+
       _updateSimpleSearchFilter(
         beforeOption: const None(),
-        positionOption: option(searchEmailFilter.value.sortOrderType.isScrollByPosition(), 0)
+        positionOption: option(searchEmailFilter.value.sortOrderType.isScrollByPosition(), 0),
       );
-      consumeState(_refreshChangesSearchEmailInteractor.execute(
+
+      final searchViewState = await _refreshChangesSearchEmailInteractor.execute(
```

Comment:
can we use refresh then merge search result with email/change?

Reply:
No, because we don't know what changed from that email change list. On the other hand the email list from search is searched with different filters.